File: src/main/resources/static/js/booky.js
document.addEventListener("DOMContentLoaded", () => {
    loadBooks();
});

function loadBooks() {
    fetch('/Booky/api/books')
        .then(response => response.json())
        .then(data => {
            const bookList = document.getElementById('bookList');
            bookList.innerHTML = '';
            data.forEach(book => {
                const bookDiv = document.createElement('div');
                bookDiv.classList.add('book');
                bookDiv.innerHTML = `
                    <img src="${book.imageUrl}" alt="${book.title}" class="book-image">
                    <h3>${book.title}</h3>
                    <p>Autor: ${book.author}</p>
                    <p>Preis: ${book.price} €</p>
                    <p>${book.description}</p>
                    <p>${book.sellerProfile}</p>
                    <button onclick="buyBook(${book.id})" class="buy-button">Kaufen</button>
                `;
                bookList.appendChild(bookDiv);
            });
        })
        .catch(error => console.error('Fehler beim Laden der Bücher:', error));
}

function searchBooks() {
    const query = document.getElementById('searchInput').value;
    fetch(`/Booky/api/books/search?query=${query}`)
        .then(response => response.json())
        .then(data => {
            const bookList = document.getElementById('bookList');
            bookList.innerHTML = '';
            data.forEach(book => {
                const bookDiv = document.createElement('div');
                bookDiv.classList.add('book');
                bookDiv.innerHTML = `
                    <img src="${book.imageUrl}" alt="${book.title}" class="book-image">
                    <h3>${book.title}</h3>
                    <p>Autor: ${book.author}</p>
                    <p>Preis: ${book.price} €</p>
                    <p>${book.description}</p>
                    <p>${book.sellerProfile}</p>
                    <button onclick="buyBook(${book.id})" class="buy-button">Kaufen</button>
                `;
                bookList.appendChild(bookDiv);
            });
        })
        .catch(error => console.error('Fehler bei der Suche nach Büchern:', error));
}

function buyBook(bookId) {
    alert(`Buch mit der ID ${bookId} wurde gekauft!`);
}

File: src/main/java/de/htwberlin/webtech/Webtech_projekt/Booky.java
package de.htwberlin.webtech.Webtech_projekt;

public class Booky {
    private int id;
    private String title;
    private String author;
    private String imageUrl;
    private double price;
    private String description;
    private String sellerProfile;

    public Booky(int id, String title, String author, String imageUrl, double price, String description, String sellerProfile) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.imageUrl = imageUrl;
        this.price = price;
        this.description = description;
        this.sellerProfile = sellerProfile;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public double getPrice() {
        return price;
    }

    public String getDescription() {
        return description;
    }

    public String getSellerProfile() {
        return sellerProfile;
    }
}
